### ISSUE TITLE
Poor performance with 3840x2160 (UHD) MP4 files

### ISSUE THREAD

**Issue by Mgiv64** [saiarcot895/chromium-ubuntu-build#33]
Hello Saikrishna, Hello All,

I've been able to get GPUVideoDecoder working fine on your version of Chromium from 66 to 70.
Checked thanks to "GPUVideoDecoder" in the chrome://media-internals and I can also see that the CPU usage is drastically lower than usual) I'm very grateful for your efforts! Thank you!

I would like to have your advice (or other users's experience) about the following problem: **When playing UHD/4K mp4 video playback remains jerky / not smooth.** 

Here is my setup:
Intel NUC 7i5 (Core i5-7300U Processor (3M Cache, up to 3.50 GHz))
4 GB RAM
GPU: Intel HD Graphics 620 (max res: 4096x2304)
Xubuntu 18.04 LTS (also tested with Ubuntu 17.10 and 18.04 LTS, same result)

To check it's was not about a hardware limitation problem, I've also tried to install Windows on the same device with official Google Chrome and 4K videos are perfectly smooth even @60 FPS 

Do you have any idea what factor could result in this loss of performance on the GNU/Linux version? is it about the Chromium implementation which is not optimal? is it about the linux drivers? something else? Does someone has been able to get a smooth playback of 4K video, maybe with another GPU? Do you have any advice what I can try to tune on the system to get smooth 4K video.

Any information/feedback/advice would be very much appreciated. Thanks.

**Comment by saiarcot895** [comment]
Just to check, is it still using the GPU video decoder with 4K videos?

**Comment by Mgiv64** [comment]
Yes, it is still using GPUVideoDecoder with 4K video. (it has to be MP4/H.264 instead of MKV/VP9).
The image is smoother than with no GpuVideoDecoder and the CPU usage is much lower, however, it is not smooth.

**Comment by Mgiv64** [comment]
another information: I've just tested video playback with VLC with VA-API Hardware Acceleration enable and the video is very smooth, so the hardware seems capable.
